packages/perseus/src/validation.types.ts

```diff
@@ -236,58 +238,105 @@ export type PerseusTableRubric = {
+    // interaction: PerseusInteractionScoringData;
+    "interactive-graph": PerseusInteractiveGraphRubric;
+    "label-image": PerseusLabelImageRubric;
+    matcher: PerseusMatcherRubric;
+    matrix: PerseusMatrixRubric;
+    // measurer: PerseusMeasurerRubric;
+    // "molecule-renderer": PerseusMoleculeRendererRubric;
+    "number-line": PerseusNumberLineScoringData;
+    "numeric-input": PerseusNumericInputRubric;
+    orderer: PerseusOrdererRubric;
+    // "passage-ref-target": PerseusRefTargetRubric;
+    // "passage-ref": PerseusPassageRefRubric;
+    // passage: PerseusPassageRubric;
+    // "phet-simulation": PerseusPhetSimulationRubric;
+    // "python-program": PerseusPythonProgramRubric;
+    plotter: PerseusPlotterScoringData;
+    radio: PerseusRadioRubric;
+    sorter: PerseusSorterRubric;
+    table: PerseusTableRubric;
+    // video: PerseusVideoRubric;
+
+    // Deprecated widgets
+    // sequence: PerseusAutoCorrectRubric;
+}
+
+/**
+ * A map of scoring data (previously referred to as "rubric"), keyed by
+ * `widgetId`. This data is used to score a learner's guess for a PerseusItem.
+ *
+ * NOTE:  The value in this map is intentionally a subset of WidgetOptions<T>.
+ * By using the same shape (minus any unneeded render data), we are able to
+ * share functionality that understands how to traverse maps of `widget id` to
+ * `options`.
+ */
+export type RubricMap = {
+    [Property in keyof RubricRegistry as `${Property} ${number}`]: {
+        type: Property;
+        static?: boolean;
+        options: RubricRegistry[Property];
+    };
+};
+
+export type Rubric = RubricRegistry[keyof RubricRegistry];
+
+// This is an interface so that it can be extended if a widget is created
+// outside of this Perseus package. See `PerseusWidgetTypes` for a full
+// explanation.
+interface UserInputRegisry {
+    categorizer: PerseusCategorizerUserInput;
+    "cs-program": PerseusCSProgramUserInput;
+    dropdown: PerseusDropdownUserInput;
+    expression: PerseusExpressionUserInput;
+    grapher: PerseusGrapherUserInput;
+    group: PerseusGroupUserInput;
+    iframe: PerseusIFrameUserInput;
+    "input-number": PerseusInputNumberUserInput;
+    "interactive-graph": PerseusInteractiveGraphUserInput;
+    "label-image": PerseusLabelImageUserInput;
+    matcher: PerseusMatcherUserInput;
+    matrix: PerseusMatrixUserInput;
+    "number-line": PerseusNumberLineUserInput;
+    "numeric-input": PerseusNumericInputUserInput;
+    orderer: PerseusOrdererUserInput;
+    plotter: PerseusPlotterUserInput;
+    radio: PerseusRadioUserInput;
+    sorter: PerseusSorterUserInput;
+    table: PerseusTableUserInput;
+}
+
+// A union type of all the widget user input types
+export type UserInput = UserInputRegisry[keyof UserInputRegisry];
+
+/**
+ * A map of widget IDs to user input types (strongly typed based on the format
+ * of the widget ID).
+ */
+export type UserInputMap = MakeWidgetMap<UserInputRegisry>;
```

Comment:
Do you only use the asterisk comments in certain situations? Just wondering when it should be the `//` version or the `/**` version.

Reply:
Oops, I should be consistent. I use `/** */` for comments that would be useful to see if I used the type somewhere else. 

Eg. 
<img width="615" alt="image" src="https://github.com/user-attachments/assets/312f27c0-3d25-4f4b-a60a-9337a404ebca" />